### ISSUE TITLE
Service Binding Operator does not detect changes to annotation information sources and push to bound apps

### ISSUE THREAD

**Issue by jeremyary** [redhat-developer/service-binding-operator#992]
### What is the environment (Minikube, Openshift)?
Openshift

### What is the SBO version used?
0.8.0

### What are the steps to reproduce this issue?
  1. Define annotations on a CRD that points to a secret and/or configmap as information source(s):
  ```
    service.binding/credentials: 'path={.status.credentialsRef},objectType=Secret'
    service.binding/configuration: 'path={.status.connectionInfo},objectType=ConfigMap'
  ```
  2. Create a CR instance with the appropriately mapped properties as well as the secret/configmap resources indicated by the annotations from step 1
  3. Create an application deployment to serve as the binding target 
  4. Create a ServiceBinding resource that binds the CR from step 2 to the target application from step 3
  5. Verify that values from the information sources (secret and configmap) are injected into the target application as a file mount
  6. Change values on the secret and configmap that the annotations point to
  7. Check the values in the target application's file mount again after allowing enough time for a potential rebind or update to occur
  8. Note that values inside the target application's file mount still reflect the old values of the configMap & secret from step 2 instead of the updated values as changed in step 6

### What is the expected behaviour?
  1. Service Binding Operator detects that an information source used to provide binding values has been changed
  2. SBO identifies resources that are already bound to the information source
  3. SBO updates the binding resource between them & the resources used to store the bound values
  4. the target bound application is bounced, thus receiving the updated values

### What is the actual behaviour?
After the initial bind, changing an information source defined via annotation does not get picked up by SBO, so values are not updated in bound applications.

### Service Binding Operator Logs
N/A

### Additional Information (Screenshots, etc)

**Comment by pedjak** [comment]
> Service Binding Operator detects that an information source used to provide binding values has been changed

In order to achieve this behavior, the operator would need to watch for changes for almost all resources in clusters, and this is very costly. Early in the development, we tried to do exactly that and were hit back with all sorts of resource utilization issues. Eventually we gave up in https://github.com/redhat-developer/service-binding-operator/pull/903, reaching excellent performances/resource utilization/stability, see [the graphs](https://github.com/redhat-developer/service-binding-operator/pull/903#issuecomment-803969568). There is still https://github.com/redhat-developer/service-binding-operator/issues/846 where we would like to see if we could still reintroduce resource watching but this time on a way lower scale.

When using annotations, SBO collects binding data from the sources and then creates another binding secret that gets injected into the application resource.

Meanwhile, there is a way to use source secrets and inject them into the application. With that any change on the secret is going to be propagated to the application pod. This approach is called [provisioned service](https://github.com/k8s-service-bindings/spec#provisioned-service) and it is recommended by the spec as the way to expose binding data.
